projects/ng-wcpms/src/lib/ng-wcpms.component.ts

```diff
@@ -0,0 +1,218 @@
+    pos_t_plus.setDate(pos_t_plus.getDate() + 5);
+
+    this.data_plotly = [];
+
+    var trace1 = {
+      mode: "lines",
+      name:  this.band,
+      x: timeline,
+      y: timeseries,
+      line: {color: '#17BECF'}
+    }
+
+    var trace2 = {
+      mode: "lines",
+      name:  "Smooth " + this.band,
+      x: timeline,
+      y: this.smooth(timeseries, 0.85),
+      line: {color: '#ff0000'}
+    }
+
+    var trace3 = {
+      mode: "markers",
+      name:  'SOS',
+      x: [phenometrics.sos_t.split('T')[0]],
+      y: [phenometrics.sos_v],
+      marker: {color: '#008c00', size: 12, line: {color: '#000000', width: 2 }},
+    }
+
+    var trace4 = {
+      mode: "markers",
+      name:  'POS',
+      x: [phenometrics.pos_t.split('T')[0]],
+      y: [phenometrics.pos_v],
+      marker: {color: '#0009e3', size: 12, line: {color: '#000000', width: 2 }}
+    }
+
+    var trace5 = {
+      mode: "markers",
+      name:  'EOS',
+      x: [phenometrics.eos_t.split('T')[0]],
+      y: [phenometrics.eos_v],
+      marker: {color: '#8a6100', size: 12, line: {color: '#000000', width: 2 }}
+    }
+
+    var trace6 = {
+      mode: "markers",
+      name:  'VOS',
+      x: [phenometrics.vos_t.split('T')[0]],
+      y: [phenometrics.vos_v],
+      marker: {color: '#e35400', size: 12, line: {color: '#000000', width: 2 }}
+    }
+
+    var trace7 = {
+      mode: "lines",
+      name:  "LOS",
+      x: [phenometrics.sos_t.split('T')[0], phenometrics.eos_t.split('T')[0]],
+      y: [phenometrics.sos_v, phenometrics.eos_v],
+      showlegend: false,
+      line: {color: '#000000', dash: 'dashdot'}
+    }
+
+    var trace8 = {
+      mode: "lines",
+      name:  "AOS",
+      x: [phenometrics.pos_t.split('T')[0], phenometrics.pos_t.split('T')[0]],
+      y: [phenometrics.pos_v, 0],
+      showlegend: false,
+      line: {color: '#000000', dash: 'dashdot'}
+    }
+
+    var trace9 = {
+      mode: "lines",
+      name:  "LIOS",
+      x: [phenometrics.sos_t.split('T')[0], phenometrics.sos_t.split('T')[0], phenometrics.pos_t.split('T')[0], phenometrics.eos_t.split('T')[0], phenometrics.eos_t.split('T')[0]],
+      y: [0, phenometrics.sos_v, phenometrics.pos_v, phenometrics.eos_v, 0],
+      fill: 'toself',
+      showlegend: false,
+      fillcolor: 'rgba(153, 247, 254, 0.4)',
+      line: {color: 'rgba(153, 247, 254, 0.4)'}
+    }
+
+    var trace10 = {
+      mode: "lines",
+      name:  "ROI",
+      x: [phenometrics.sos_t.split('T')[0], pos_t_minus],
+      y: [phenometrics.sos_v+1500, phenometrics.pos_v+1500],
+      showlegend: false,
+      line: {color: '#000000', dash: 'dashdot'}
+    }
+
+    var trace11 = {
+      mode: "lines",
+      name:  "ROD",
+      x: [pos_t_plus, phenometrics.eos_t.split('T')[0]],
+      y: [phenometrics.pos_v+1500, phenometrics.eos_v+1500],
+      showlegend: false,
+      line: {color: '#000000', dash: 'dashdot'}
+    }
+
+    this.data_plotly = [trace9, trace1, trace2, trace7, trace8, trace3, trace4, trace5, trace6];
```

Comment:
Why not create a key-value object to control these plots?
There are variables like “trace10” and “trace11” that are not being used.